src/python/pants/backend/go/util_rules/embed_integration_test.py

```diff
@@ -234,3 +236,53 @@ def test_embed_in_external_test(rule_runner: RuleRunner) -> None:
     tgt = rule_runner.get_target(Address("", target_name="pkg"))
     result = rule_runner.request(TestResult, [GoTestFieldSet.create(tgt)])
     assert result.exit_code == 0
+
+
+@logging
+def test_third_party_package_embed(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "BUILD": dedent(
+                """
+                go_mod(name='mod')
+                go_package(name='pkg')
+                """
+            ),
+            "go.mod": dedent(
+                """\
+                module go.example.com/foo
+                go 1.17
+
+                require (
+                \tgithub.com/tdyas/go-embed-sample-for-test v0.0.2
```

Comment:
This repo will need to be moved into `pantsbuild` GitHub org.

Reply:
Would it be possible to treat this as a vendored dep somehow, and then put it in a tarball in `pantsbuild/pants`? Or does this test require `git` to be in the mix?

Reply:
I don't know offhand. The Go backend does not currently support vendored sources (under a `vendor/` directory). Searching finds [this comment on Reddit](https://www.reddit.com/r/golang/comments/ejj6xm/how_to_use_module_from_tarball/fd42znn/?utm_source=reddit&utm_medium=web2x&context=3) which proposes storing the module zip file and metadata in a directory that is then listed in the `GOPROXY` environment variable.

Reply:
I don't believe `git` is required here since `go` contacts the official Go module proxy generally to download modules. (The official module proxy may use `git` as an implementation detail but that isn't exposed to clients.)

Reply:
Latest commit makes everything into a local Go module proxy. So the test doesn't need an external download.